### PR TITLE
Added Google Analytics Code for Amy Bot

### PR DESCRIPTION
**Pull request by maqankitat**
Google Analytics Code for Amy

### CODE REVIEW

File: css/bot.css [hunk 1]
```diff
@@ -688,7 +687,7 @@ body .wc-app, .wc-app button, .wc-app input, .wc-app textarea {
 	}
 
 		.wc-console .wc-textbox input {
-			padding: 0px;
+			
 			background-color: #ffffff;
```
Comment (MAQHatimD) [inline css/bot.css]: use #fff instead of #ffffff

File: js/renderBot.js [hunk 2]
```diff
@@ -166,12 +166,43 @@ function renderBot(directLineKey, botSecret) {
         }
     });
 
+    // calling session storage data to other pages on Bot 
+    $(function storeddata() {
+        if (sessionStorage["myKey"] != null) {
```
Comment (MAQHatimD) [inline js/renderBot.js]: keep constant on left hand side of comparison

File: js/renderBot.js [hunk 3]
```diff
@@ -166,12 +166,43 @@ function renderBot(directLineKey, botSecret) {
...
+            var contentsOfOldDiv = JSON.parse(sessionStorage["myKey"]);
+            $(".wc-message-group-content").html(contentsOfOldDiv);
+        }
+    });
+
+    $(function appendmessage() {
+        var a;
```
Comment (MAQHatimD) [inline js/renderBot.js]: provide descriptive name to variable

File: js/renderBot.js [hunk 4]
```diff
@@ -166,12 +166,43 @@ function renderBot(directLineKey, botSecret) {
...
+        newMessages = document.getElementsByClassName("wc-message-groups")[0];
```
Comment (MAQHatimD) [inline js/renderBot.js]: check for nullability

File: js/renderBot.js [hunk 5]
```diff
@@ -166,12 +166,43 @@ function renderBot(directLineKey, botSecret) {
...
+        newMessages.addEventListener('DOMNodeInserted', function (evt) {
+            if (newMessages.getElementsByClassName("wc-message-group-content") !== undefined) {
+                var msgList = document.getElementsByClassName('wc-message-wrapper list').length;
```
Comment (MAQHatimD) [inline js/renderBot.js]: Use either Jquery or JavaScript consistently

File: js/renderBot.js [hunk 6]
```diff
@@ -166,12 +166,43 @@ function renderBot(directLineKey, botSecret) {
...
+                if (isFirstMessage === 0 && $(evt.target).attr('class') === 'wc-message-wrapper list') {
+                    isFirstMessage = 1;
+                    var targetNode = document.getElementsByClassName('wc-message-wrapper list')[0];
+                    newMessages.getElementsByClassName("wc-message-group-content")[0].removeChild(targetNode);
+                    newMessages.getElementsByClassName("wc-message-group-content")[0].appendChild(targetNode);
```
Comment (MAQHatimD) [inline js/renderBot.js]: Use jquery chaining

File: js/renderBot.js [hunk 7]
```diff
@@ -166,12 +166,43 @@ function renderBot(directLineKey, botSecret) {
...
+            }
+        });
+    });
+
+
     typingActions = document.getElementsByClassName("wc-message-groups")[0];
     typingActions.addEventListener('DOMNodeInserted', function () {
         if (typingActions.getElementsByClassName("wc-typing")[0] !== undefined) {
             $(".wc-typing").parent().css("background-color", "transparent");
             $(".wc-typing").parent().css("border-radius", "1px");
```
Comment (MAQHatimD) [inline js/renderBot.js]: Use css class

File: css/bot.css (hunk 1 above)
Comment (maqankitat) [reply css/bot.css]: Changed made accordingly

File: js/renderBot.js (hunk 2 above)
Comment (maqankitat) [reply js/renderBot.js]: Resolved comments accordindly

File: js/renderBot.js (hunk 3 above)
Comment (maqankitat) [reply js/renderBot.js]: changes done as stated

File: js/renderBot.js (hunk 4 above)
Comment (maqankitat) [reply js/renderBot.js]: Condition invoked as stated

File: js/renderBot.js (hunk 5 above)
Comment (maqankitat) [reply js/renderBot.js]: Used JavaScript and changed accordingly

File: js/renderBot.js (hunk 6 above)
Comment (maqankitat) [reply js/renderBot.js]: Chaining won't work in the case of remove and append.
chaning error

File: js/renderBot.js (hunk 7 above)
Comment (maqankitat) [reply js/renderBot.js]: Chaining done

File: css/bot.css [hunk 8]
```diff
@@ -583,6 +583,15 @@ body .wc-app, .wc-app button, .wc-app input, .wc-app textarea {
...
 
+.wc-message-contentnew{
+background-color: #E5E5E5;
+border-radius: 10px;
+border-bottom-left-radius: 0px;
+margin-left: 8px;
+color: #5B5B5B;
+
+}
+
```
Comment (maqankitat) [inline css/bot.css]: added a new class instead of hard coded css

File: css/bot.css [hunk 9]
```diff
@@ -734,6 +743,11 @@ body .wc-app, .wc-app button, .wc-app input, .wc-app textarea {
     
 }
 
+.wc-typingmsg {
+    background-color: transparent !important ;
+    border-radius:1px !important;
+    box-shadow:none;
+}
```
Comment (maqankitat) [inline css/bot.css]: added a new class instead of hard coded css

File: js/renderBot.js [hunk 10]
```diff
@@ -157,9 +157,9 @@ function renderBot(directLineKey, botSecret) {
 
     suggestedActions = document.getElementsByClassName("wc-suggested-actions")[0];
     suggestedActions.addEventListener('DOMNodeInserted', function () {
-        if (suggestedActions.getElementsByClassName("wc-hscroll")[0] !== undefined) {
+        if (undefined !== suggestedActions.getElementsByClassName("wc-hscroll")[0]) {
```
Comment (maqankitat) [inline js/renderBot.js]: constants values shifted to left hand side

File: js/renderBot.js [hunk 11]
```diff
@@ -168,64 +168,41 @@ function renderBot(directLineKey, botSecret) {
...
+                if (undefined !== newMessages.getElementsByClassName("wc-message-group-content")) {
+                    var msgList = document.getElementsByClassName('wc-message-wrapper list').length;
+                    if (isFirstMessage === 0 && evt.target.getAttribute('class') === 'wc-message-wrapper list') {
+                        isFirstMessage = 1;
+                        var targetNode = document.getElementsByClassName('wc-message-wrapper list')[0];
+                        var grpContent = newMessages.getElementsByClassName("wc-message-group-content")[0];
+                        grpContent.removeChild(targetNode);
+                        grpContent.appendChild(targetNode);
+
+                    }
```
Comment (maqankitat) [inline js/renderBot.js]: did the changes accordingly

**Pull request merged**